src/serialization/binary_archive.h

```diff
@@ -145,9 +141,11 @@ struct binary_archive<false> : public binary_archive_base<std::istream, false>
   template <class T>
   void serialize_uvarint(T &v)
   {
-    typedef std::istreambuf_iterator<char> it;
-    if (tools::read_varint(it(stream_), it(), v) < 0)
-      stream_.setstate(std::ios_base::failbit);
+    auto current = bytes_.begin();
+    auto end = bytes_.end();
```

Comment:
Minor, but could it be:

`const auto end = bytes_.cend();`

and then reused at lines 147 and 148?

Reply:
Changed to `cbegin()`/`cend()`.

I would prefer to make line 145 `const auto end = bytes_.cend()`, but the types have to match exactly. And I don't want `end` to be modified by the function call (it doesn't, but another call to `cend()` guarantees the value is not modified).